File: image/src/main/java/com/trailmagic/util/SecurityUtil.java
package com.trailmagic.util;

import com.trailmagic.user.User;
import com.trailmagic.user.security.ToolkitUserDetails;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContext;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Service;

@Service("securityUtil")
public class SecurityUtil {
    public User getCurrentUser() {
        SecurityContext securityContext = SecurityContextHolder.getContext();
        final Authentication authentication = securityContext.getAuthentication();
        if (authentication == null) {
            return null;
        }

        ToolkitUserDetails userDetails = (ToolkitUserDetails) authentication.getPrincipal();
        if (userDetails == null) {
            return null;
        }

        return userDetails.getRealUser();
    }
}

File: image/src/main/scala/com/trailmagic/util/SecurityUtil.scala
package com.trailmagic.util

import com.trailmagic.user.User
import com.trailmagic.user.security.ToolkitUserDetails
import org.springframework.security.core.Authentication
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Service

@Service("securityUtil")
class SecurityUtil {
  def getCurrentUser: User = {
    val authentication: Authentication = SecurityContextHolder.getContext.getAuthentication

    authentication match {
      case null => throw new IllegalStateException("Must have a signed-in user–no Authentication present")
      case _ => authentication.getPrincipal.asInstanceOf[ToolkitUserDetails] match {
        case null => throw new IllegalStateException("Must have a signed-in user–no principal present")
        case userDetails => userDetails.getRealUser
      }
    }
  }
}